snowpack/assets/require-or-import.js

```diff
@@ -20,16 +20,28 @@ module.exports = async function requireOrImport(filepath, { from = process.cwd()
         });
         try {
           const mdl = NATIVE_REQUIRE(resolvedPath);
+
+          // Add a `default` property on a CommonJS export so that it can be accessed from import statements.
+          // This is set to enumerable: false to make it hidden(ish) to code.
+          if(mdl && !('default' in mdl)) {
+            Object.defineProperty(mdl, 'default', {
+              configurable: true,
+              enumerable: false,
+              writable: true,
+              value: mdl
+            });
+          }
+
           resolve(mdl);
         } catch (e) {
           if (e instanceof SyntaxError && /export|import/.test(e.message)) {
             console.error(`Failed to load "${filepath}"!\nESM format is not natively supported in "node@${process.version}".\nPlease use CommonJS or upgrade to an LTS version of node above "node@12.17.0".`)
           } else if (e.code === 'ERR_REQUIRE_ESM') {
               const url = pathToFileURL(resolvedPath);
-              return NATIVE_IMPORT(url).then(mdl => resolve(mdl.default ? mdl.default : mdl));
-          };
+              return NATIVE_IMPORT(url).then(mdl => resolve(mdl.default ? mdl.default : mdl), err => reject(err));
```

Comment:
Is there any difference between onRejected vs `catch()`? I have a slight preference for the latter because it’s easier to read but don’t feel too strongly

Reply:
.catch matters where you put it, if you put it after the then it would reject if the caller had an error, when is not what we want. Could have put the .catch before the then though.